public/app/features/dashboard-scene/utils/utils.ts

```diff
@@ -261,3 +262,18 @@ export function getLibraryPanel(vizPanel: VizPanel): LibraryVizPanel | undefined
   }
   return;
 }
+
+export function isWithinUnactivatedRepeatRow(panel: VizPanel): boolean {
```

Comment:
```suggestion
/**
* If the panel is within a repeated row, it must wait until the row resolves the variables.
* This ensures that the scoped variable for the row is assigned and the panel is initialized with them.
*/
export function isWithinUnactivatedRepeatRow(panel: VizPanel): boolean {
```
Just adding some extra context :)

Reply:
That's a very good point! Thank you for adding this, I'm committing as we speak!